src/System Application/App/Azure AD Plan/src/UserPlanExperience.Enum.al

```diff
@@ -0,0 +1,46 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+
+namespace System.Azure.Identity;
+
+/// <summary>
+/// User plan experience. 
+/// </summary>
+enum 9005 "User Plan Experience"
+{
+    Extensible = false;
+
+    /// <summary>
+    /// Essentials plan.
+    /// </summary>
+    value(0; Essentials)
+    {
+        Caption = 'Essentials';
+    }
+
+    /// <summary>
+    /// Premium plan.
+    /// </summary>
+    value(2; Premium)
+    {
+        Caption = 'Premium';
+    }
+
+    /// <summary>
+    /// Basic plan.
+    /// </summary>
+    value(3; Basic)
+    {
+        Caption = 'Basic';
+    }
```

Comment:
Why not have all possible plans? Team Member etc. could come in handy too given time, could they not?

Reply:
The purpose of the enum was to return only the information relevant for the mixed plans check, hence the name "User Plan Experience". Currently we're only checking mixed plans for Basic, Essentials and Premium licenses which is why these are the plans represented in the enum. Adding all plans / roles to this enum would add unnecessary overhead to the login flow. 

Furthermore, if we start considering all plans / roles, a single enum would not be sufficient to accurately represent the plans of a user, since one user can have more than one "plan".

So resolving this comment